### PR TITLE
mon: clean up cluster logging on mon events

### PR DESCRIPTION
**Pull request by jcsp**
Fixes: http://tracker.ceph.com/issues/22082

This is several minor tweaks to make sure what comes out the cluster log is sane and readable when a mon goes down.

### CODE REVIEW

File: src/mon/Monitor.cc [hunk 1]
```diff
@@ -1918,8 +1918,20 @@ void Monitor::win_election(epoch_t epoch, set<int>& active, uint64_t features,
   pending_metadata = metadata;
   outside_quorum.clear();
 
-  clog->info() << "mon." << name << "@" << rank
-		<< " won leader election with quorum " << quorum;
+  std::stringstream msg;
+  msg << "mon." << name << " is new leader, mons ";
+  for (auto i = quorum.begin(); i != quorum.end();) {
+    derr << "rank: " << *i << dendl;
```
Comment (liewegas) [inline src/mon/Monitor.cc]: drop this

Comment (liewegas) [reply src/mon/Monitor.cc]: also, see Monitor::get_quorum_names()?

Comment (jcsp) [reply src/mon/Monitor.cc]: Thanks, I had a feeling I was forgetting about a helper

**Comment by jcsp** [comment]
Updated

File: src/mon/Monitor.h [hunk 2]
```diff
@@ -635,8 +635,8 @@ class Monitor : public Dispatcher,
     return (class MgrStatMonitor*) paxos_service[PAXOS_MGRSTAT];
   }
 
-  class MgrStatMonitor *healthmon() {
-    return (class MgrStatMonitor*) paxos_service[PAXOS_MGRSTAT];
+  class HealthMonitor *healthmon() {
```
Comment (tchaikov) [inline src/mon/Monitor.h]: is this method used anywhere?

Comment (jcsp) [reply src/mon/Monitor.h]: it wasn't before, but it is now in Monitor::win_election

Comment (tchaikov) [reply src/mon/Monitor.h]: ahh, i see.

File: src/mon/Monitor.cc [hunk 3]
```diff
@@ -1960,7 +1960,26 @@ void Monitor::win_election(epoch_t epoch, set<int>& active, uint64_t features,
...
+      if (healthmon()->is_proposing()) {
+        dout(4) << __func__ << " healthmon proposing, waiting" << dendl;
+        healthmon()->wait_for_finished_proposal(nullptr, new C_MonContext(this,
+              [this](int r){
+                assert(lock.is_locked_by_me());
+                do_health_to_clog_interval();
+              }));
+
+      } else {
+        dout(4) << __func__ << " healthmon is not proposing" << dendl;
```
Comment (tchaikov) [inline src/mon/Monitor.cc]: @jcsp can we lower the debug level to, for example, 10?

Comment (jcsp) [reply src/mon/Monitor.cc]: Done (actually removed this one and demoted the one above to 20, I don't think we need the logging on the non-waiting case.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2017-11-10_06:12:07-rados-wip-kefu-testing-2017-11-10-1154-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2017-11-12_17:42:28-rados-wip-kefu-testing-2017-11-10-1154-distro-basic-mira/

**Pull request merged**